src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -631,25 +630,36 @@ static bool _rasterLinearGradientRect(SwSurface* surface, const SwBBox& region,
     } else {
         if (surface->compositor) {
             auto method = surface->compositor->method;
-            uint32_t *shape = fill->ctable;
             auto cbuffer = surface->compositor->image.data + (region.min.y * surface->stride) + region.min.x;
-            auto sbuffer = shape + (region.min.y ) + region.min.x;
-            for (uint32_t y = 0; y < h; ++y) {
-                auto dst = &buffer[y * surface->stride];
-                auto cmp = &cbuffer[y * surface->stride];
-                auto src = &sbuffer[y];
-                if (method == CompositeMethod::AlphaMask) {
-                    for (uint32_t x = 0; x < w; ++x, ++dst, ++cmp) {
+            auto sbuffer = static_cast<uint32_t*>(alloca(w * sizeof(uint32_t)));
```

Comment:
please double check whether the sbuffer size should be 'w' and not 'surface->w' as it was originally in l.727